oak/server/rust/src/asylo_alloc.rs

```diff
@@ -0,0 +1,34 @@
+use core::alloc::{GlobalAlloc, Layout};
+
+extern "C" {
+    // See https://www.sourceware.org/newlib/libc.html#malloc.
+    fn malloc(nbytes: usize) -> *mut u8;
+    fn memalign(align: usize, nbytes: usize) -> *mut u8;
+    fn free(aptr: *mut u8);
+}
+
+// Aligment is defined in the Intel SGX SDK.
+// See https://github.com/intel/linux-sgx/blob/master/sdk/tlibc/stdlib/malloc.c#L541.
+const MIN_ALIGN: usize = 8;
+
+/// A global allocator that links directly to the primitives exposed by Asylo, which in turns relies
```

Comment:
nit: s/global allocator/memory allocator/ 

(IIUC: Despite the `GlobalAllocator` trait name, what's defined here is a type not a global allocator (yet); the point where the global allocator gets defined is the `#[global_allocator]` static instance in the other file.)